Fails to build with Boost 1.70 (Beta1)

Regressed by boostorg/core@d8895bab4676:
```c++
In file included from segmentation/src/supervoxel_clustering.cpp:49:
In file included from segmentation/include/pcl/segmentation/impl/supervoxel_clustering.hpp:43:
segmentation/include/pcl/segmentation/supervoxel_clustering.h:527:26: error: 'checked_delete<const
      pcl::SupervoxelClustering<pcl::PointXYZRGB>::SupervoxelHelper>' is missing exception specification
      'noexcept'
  ...boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *);
            ^
                                                                                                    noexcept
segmentation/include/pcl/segmentation/impl/supervoxel_clustering.hpp:704:75: note: in instantiation of template
      class 'pcl::SupervoxelClustering<pcl::PointXYZRGB>' requested here
    pcl::octree::OctreePointCloudAdjacencyContainer<pcl::PointXYZRGB,pcl::SupervoxelClustering<pcl::PointX...
                                                                          ^
segmentation/include/pcl/segmentation/supervoxel_clustering.h:527:26: note: previous declaration is here
      friend void boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *);
                         ^
segmentation/include/pcl/segmentation/supervoxel_clustering.h:527:26: error: 'checked_delete<const
      pcl::SupervoxelClustering<pcl::PointXYZRGBA>::SupervoxelHelper>' is missing exception specification
      'noexcept'
  ...boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *);
            ^
                                                                                                    noexcept
segmentation/include/pcl/segmentation/impl/supervoxel_clustering.hpp:708:76: note: in instantiation of template
      class 'pcl::SupervoxelClustering<pcl::PointXYZRGBA>' requested here
    pcl::octree::OctreePointCloudAdjacencyContainer<pcl::PointXYZRGBA,pcl::SupervoxelClustering<pcl::Point...
                                                                           ^
segmentation/include/pcl/segmentation/supervoxel_clustering.h:527:26: note: previous declaration is here
      friend void boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *);
                         ^
segmentation/include/pcl/segmentation/supervoxel_clustering.h:527:26: error: 'checked_delete<const
      pcl::SupervoxelClustering<pcl::PointXYZ>::SupervoxelHelper>' is missing exception specification 'noexcept'
  ...boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *);
            ^
                                                                                                    noexcept
segmentation/include/pcl/segmentation/impl/supervoxel_clustering.hpp:720:72: note: in instantiation of template
      class 'pcl::SupervoxelClustering<pcl::PointXYZ>' requested here
    pcl::octree::OctreePointCloudAdjacencyContainer<pcl::PointXYZ,pcl::SupervoxelClustering<pcl::PointXYZ>...
                                                                       ^
segmentation/include/pcl/segmentation/supervoxel_clustering.h:527:26: note: previous declaration is here
      friend void boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *);
                         ^
3 errors generated.
```

--- a/segmentation/include/pcl/segmentation/supervoxel_clustering.h
+++ b/segmentation/include/pcl/segmentation/supervoxel_clustering.h
@@ -40,6 +40,8 @@
 
 #pragma once
 
+#include <boost/version.hpp>
+
 #include <pcl/features/normal_3d.h>
 #include <pcl/pcl_base.h>
 #include <pcl/point_cloud.h>
@@ -524,7 +526,11 @@ namespace pcl
       };
 
       //Make boost::ptr_list can access the private class SupervoxelHelper
+#if BOOST_VERSION >= 107000
+      friend void boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *) BOOST_NOEXCEPT;
+#else
       friend void boost::checked_delete<> (const typename pcl::SupervoxelClustering<PointT>::SupervoxelHelper *);
+#endif
 
       typedef boost::ptr_list<SupervoxelHelper> HelperListT;
       HelperListT supervoxel_helpers_;

